### PR TITLE
Rewrite kubeflow-volumes-operator as sidecar

### PR DESCRIPTION
**Pull request by misohu**
In this PR:
- Rewrote the operator using BaseCharm sidecar pattern. 
- Rewrote unit tests to cover the new source code. 
- The kubeflow manifests for cluster roles are now part of this charm. Make sure to approve this [PR](https://github.com/canonical/kubeflow-roles-operator/pull/78) for kubeflow-roles-operator as the manifests are no longer needed there (previously we were not able to deploy those with podspec charm).

Because there are no proper integration tests as mentioned in https://github.com/canonical/kubeflow-volumes-operator/issues/29, to test this PR I have deployed kubeflow-bundle 1.8 with this charm and I made sure I am able to 
- Create a new volume in the volumes UI.
- I am able to use the newly created volume from inside the UI.
- I am able to remove this volume from UI.

CI note: publish is failing because I renamed the resource in charm. Will publish new oci image when approved <3

### CODE REVIEW

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -51,14 +50,6 @@ async def test_build_and_deploy(ops_test: OpsTest):
     )
 
 
-@pytest.mark.abort_on_fail
-async def test_configmap_created(lightkube_client: Client, ops_test: OpsTest):
```
Comment (orfeas-k) [inline tests/integration/test_charm.py:55]: Why did we remove this test?

**Review by orfeas-k — COMMENTED** [review]
Thank you @misohu. I have left some comments. I see that you followed patterns from other charms in implementing the CharmReconciler which helps with reviewing too, since those patterns have been already tested. I have a question though, did you use `kubeflow-roles` from [PR](https://github.com/canonical/kubeflow-roles-operator/pull/78) when testing the charm?

File: config.yaml
[the remark this answers is not in the document]
Comment (misohu) [reply config.yaml:1]: So Lets do It in separate PR so we can easily rollback in case of trouble :) WDYT?

[the remark this answers is not in the document]
Comment (orfeas-k) [reply config.yaml:1]: Sure, this makes sense actually

**Review by orfeas-k — APPROVED** [review]
Good job @misohu, approved!

**Pull request merged**